liblangutil/ErrorReporter.cpp

```diff
@@ -86,11 +86,11 @@ void ErrorReporter::error(Error::Type _type, SourceLocation const& _location, Se
 	m_errorList.push_back(err);
 }
 
-bool ErrorReporter::checkForExcessiveErrors(Error::Type _type)
+bool ErrorReporter::checkForExcessiveErrors(Error::Type _type, bool _advance)
 {
 	if (_type == Error::Type::Warning)
 	{
-		m_warningCount++;
+		if (_advance) m_warningCount++;
 
 		if (m_warningCount == c_maxWarningsAllowed)
```

Comment:
Don't you just want to query whether we are already past the limit without any side-effects?

Reply:
This still does not properly split between just querying and having side-effects in addition to just querying.